testutil/exec.go

```diff
@@ -0,0 +1,124 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"gopkg.in/check.v1"
+)
+
+// MockCmd allows mocking commands for testing.
+type MockCmd struct {
+	binDir    string
+	exeFile   string
+	logFile   string
+	countFile string
+}
+
+// MockCommand adds a mocked command to PATH.
+//
+// The command logs all invocations to a dedicated log file and exits with the
+// specified code.
+func MockCommand(c *check.C, basename string, status int) *MockCmd {
+	binDir := c.MkDir()
+	exeFile := path.Join(binDir, basename)
+	logFile := path.Join(binDir, basename+".log")
+	countFile := path.Join(binDir, basename+".cnt")
+	err := ioutil.WriteFile(exeFile, []byte(fmt.Sprintf(""+
+		"#!/bin/sh\n"+
+		"echo \"$@\" >> %q\n"+
+		"exit %d\n", logFile, status)), 0700)
+	if err != nil {
+		panic(err)
+	}
+	os.Setenv("PATH", binDir+":"+os.Getenv("PATH"))
```

Comment:
I think I would slightly prefer `os.Setenv("PATH", fmt.Sprintf("%s:%s", binDir, os.Getenv("PATH")))` here but thats probably personal preference (the + and the " and the : are hard to read(er) to read for me). But super nitpick, sorry.